### ISSUE TITLE
The survey is very slow when using onTextMarkdown to replace html on questions

### ISSUE THREAD

**Issue by jmar1998** [surveyjs/survey-library#2907]
### Are you requesting a feature, reporting a bug or asking a question?

Bug

### What is the current behavior?
When i have a big survey and i use this event, is very slow to load, and when some logic (visibleIf) is applied is very slow

Like 5-10 segs to show 1 question
### What is the expected behavior?
Take a normal time

### How would you reproduce the current behavior (if this is a bug)?

Create a big survey with all the options, and add the code added on test code section

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```
surveyModel.onTextMarkdown.add(function(survey, options) {
	var str = converter.makeHtml(options.text);
	options.html = str;
});
```
Comments : 
idk how really works this events, but seems like when we set the property html on the event something happens and seems like we got several re-renderings here
### Specify your

* browser:
* browser version:
* surveyjs platform (angular or react or jquery or knockout or vue):
* surveyjs version:

**Comment by andrewtelnov** [comment]
@jmar1998 What version do you use? I remember we have optimized the number of calls for this event.

Thank you,
Andrew

**Comment by jmar1998** [comment]
@andrewtelnov the version im using is 1.8.47

**Comment by andrewtelnov** [comment]
@jmar1998 Then you have to check how many times this event is called. You can create an example in plunker, so we can take a look.

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@jmar1998 We call onTextMarkdown every time a value changed in survey, since the html could be changed because of it. I guess we need to separate text markdown and text processing.
For now, you can hash the converting results. Here is your [updated example](https://plnkr.co/edit/0jLOydLTlt6jyBcL).

Thank you,
Andrew

**Comment by jmar1998** [comment]
thanks alot @andrewtelnov , more info about my issue, as i told before my survey was 10 times bigger + single page mode.
About the code, thanks for your snippet, i also add this to my code
survey.onTextMarkdown.add(function(survey, options) {
	if (options.name == null && !options.element.isVisible) {
		return ;
	}
	var str = converter.makeHtml(options.text);
	options.html = str;
});
I notice that at least for my case whenn the attribute name was empty, this event was not needed, also for non visible events
----------------------------------------------------------------------------------------------------------------------------------------------
Only a question @andrewtelnov , idk how works all the events handling and processing the survey/questions, but maybe its good idea to process first the question (calculate visibleif), and after that do every event/process only on the visible questions
idk its this is possible, maybe could be a good approach for a better performance

**Comment by masciugo** [comment]
thank you @andrewtelnov for the hash solution. However as a remark of what you said about the survey behaviour in this [very very basic example](https://plnkr.co/edit/Y5IloFK7laNE54q5) I noticed that the onTextMarkdown hook is called 16 times for just one question without changing anything!

<img width="923" alt="image" src="https://user-images.githubusercontent.com/454321/191054234-1a24c731-fa13-496c-9688-c3a39580814c.png">

thank you very much for any improvement

**Comment by andrewtelnov** [comment]
@masciugo @jmar1998 It has been fixed by the commit above. The fix will be available in v1.9.51.

Thank you,
Andrew
